packages/dds/matrix/src/matrix.ts

```diff
@@ -529,6 +532,36 @@ export class SharedMatrix<T = any>
 
 		// Only need to store it in the snapshot if we have switched the policy already.
 		if (this.setCellLwwToFwwPolicySwitchOpSeqNumber > -1) {
+			const cellLastWriteTrackerToSummarize = new SparseArray2D<CellLastWriteTrackerItem>();
+			let count = 100000;
+			let r = this.rowNumberToClearFromTrackerMatrix;
+			// We only need to store edit info in tracker matrix for cells which are in collab window, we can discard
+			// rest of them. This can be expensive depending upon the size of matrix but we don't need to clear all
+			// of them at once. Just clear 100K cells at a time.
+			for (; r < this.rowCount && count > 0; r++) {
+				// Map the logical (row, col) to associated storage handles.
+				const rowHandle = this.rowHandles.getHandle(r);
+				if (isHandleValid(rowHandle)) {
+					for (let c = 0; c < this.colCount; c++) {
+						count--;
+						const colHandle = this.colHandles.getHandle(c);
+						if (isHandleValid(colHandle)) {
+							const item = this.cellLastWriteTracker.getCell(rowHandle, colHandle);
+							if (
+								item !== undefined &&
+								(item.seqNum > this.rows.getCollabWindow().minSeq ||
```

Comment:
I'd highly recommend you to cache the value of "this.rows.getCollabWindow().minSeq" and not call it twice on each iteration.

Reply:
I wonder why this.cols.getCollabWindow().minSeq and this.rows.getCollabWindow().minSeq could be different. But it does not matter - we can use max() of them